File: mergesort.rb
def merge(array)
	if array.length < 2 
		return array
	else 
		middle = array.length/2
		left = array[0..(middle - 1)]
		right = array[middle..-1]
		mergeSort(merge(left), merge(right))
	end 
end 

def mergeSort(left, right)
	sorted = []
	while !left.empty? && !right.empty?
		if left[0] < right[0]
			sorted.push(left.shift)
		else 
			sorted.push(right.shift)
		end 
	end 
	sorted + right + left
end 

merge([6,25, 4, 99, 9,5,1, 20, 50, 10, 17])

File: mergesort.js

function mergeSort(array){
	
	if(array.length < 2){
		return array
	} else {
		var middle = parseInt(array.length/2);
		var left = array.slice(0, middle)
		var right = array.slice(middle, array.length)
		return merge(mergeSort(left), mergeSort(right)); 
	}
}

function merge(a1, a2){
	var sortedArray = []
	while(a1.length > 0 && a2.length > 0){

		if(a2[0] <=a1[0]){
			sortedArray.push(a2.shift())
		} else {
			sortedArray.push(a1.shift())
		}
	}
	sortedArray = sortedArray.concat(a1)
	sortedArray = sortedArray.concat(a2)
	return sortedArray;
}


mergeSort([6,9,2,3,10,22,100,5,24])